### ISSUE TITLE
Document that `test/commands` and `run_test.{sh,bat}` cannot coexist

### ISSUE THREAD

**Issue by leofang** [conda-forge/conda-forge.github.io#1490]
As title. If `meta.yaml` has the `test/commands` session
```yaml
test:
  commands:
    - ...
```
but `run_test.{sh,bat}` is also provided, the commands in `test/commands` will not be run. Workarounds include
- Move all commands to `run_test.{sh,bat}` (or vice versa)
- Rename `run_test.{sh,bat}` to something else and run it explicitly

**Comment by jakirkham** [comment]
@leofang do you recall where this came up? No worries if not. Wondering if there might be a bug actually.

**Comment by leofang** [comment]
Sorry I don't remember in which feedstock I hit the issue, but I seem to remember confirming it's a known issue with either @jaimergp or @beckermr on the Gitter channel back then before filing an issue.

**Comment by beckermr** [comment]
I don't recall this one but conda build does have its quirks. Probably need a simple test case here.

**Comment by jakirkham** [comment]
Well we just recently used this with `boost-cpp` and it does seem to work there. Please see PR ( https://github.com/conda-forge/boost-cpp-feedstock/pull/123 ).

That said, have no problem believing there are other cases where it may have run into issues. Just don't know what those might be.

**Comment by jakirkham** [comment]
So I think part of the issue is not much output is produced from `run_test.*` by default. Filed issue ( https://github.com/conda/conda-build/issues/4426 ) about addressing this.

That said, there may be more to the story. Still digging in 👷

**Comment by jakirkham** [comment]
Ok it does look like there is another bug, which is causing this issue. Filed as issue ( https://github.com/conda/conda-build/issues/4427 ).

**Comment by jaimergp** [comment]
I think I saw this in the `conda-feedstock`. https://github.com/conda-forge/conda-feedstock/pull/162

**Comment by jakirkham** [comment]
Yeah have a few reproducers now. Tried submitting a fix upstream. Could use a review if you have time to look 🙂
